packages/inertia/index.d.ts

```diff
@@ -20,10 +22,17 @@ export interface Page<CustomPageProps extends PageProps = PageProps> {
 
 type VisitOptions = {
   method?: string
+  replace?: boolean
   preserveScroll?: boolean | ((props: Inertia.PageProps) => boolean)
   preserveState?: boolean | ((props: Inertia.PageProps) => boolean)
-  replace?: boolean
   only?: string[]
+  headers?: object
+  onCancelToken?: (cancelToken: CancelTokenSource) => void
+  onStart?: (visit: VisitOptions & {url: string}) => void | boolean
+  onProgress?: (progress: any) => void // Cast to any, Axios doesn't have a type definition for this
```

Comment:
I'm pretty certain this will always be a native [ProgressEvent](https://developer.mozilla.org/en-US/docs/Web/API/ProgressEvent). Does that help at all?